### PR TITLE
Update dependencies for `with-iron-session` example

### LINKED ISSUE #15033: with-iron-session example fetch function is not defined
# Bug report

## Describe the bug

I was trying the `with-iron-session` example from the canary branch, I remember a couple days ago it was working as expected, today I tried it again but now I was getting the following error:

Response:
```
{"message":"fetch is not defined"}
```
Error:
```
index.js:1 An unexpected error happened: Error: Internal Server Error
    at fetcher$ (fetchJson.js:15)
    at tryCatch (runtime.js:45)
    at Generator.invoke [as _invoke] (runtime.js:274)
    at Generator.prototype.<computed> [as next] (runtime.js:97)
    at tryCatch (runtime.js:45)
    at invoke (runtime.js:135)
    at runtime.js:145
```

I also then installed `"isomorphic-unfetch": "^3.0.0",` and imported the `fetch` function where it's required and the example  worked again as expected.

I already removed the whole example and re-installed again but that didn't fix it.

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

1. Go to the examples in the canary branch and install `with-iron-session`
2. Run the local server
3. Try to login to fetch your GitHub user, you'll get the error
4. Install `"isomorphic-unfetch": "^3.0.0",` and import it in `lib/fetchJson.js`
5. Now try to login again and the fetch function will work.

## Expected behavior

The `fetch` function from the example to work.

## System information

- OS: Windows 10
- Browser: Version 83.0.4103.116 (Official Build) (64-bit)
- Version of Next.js: 9.3.6
- Version of Node.js: v12.16.3

### PR DESCRIPTION
**Pull request by jensmeindertsma**
This PR updates the dependencies of the `with-iron-session` example. This includes bumping the version of `next` to `9.4.4` which fixes #15033. This is because Next polyfills `fetch` on the browser side since `9.1.7` but the Node.js `fetch` polyfill wasn't included until `9.4`

### CODE REVIEW

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | jensmeindertsma/next.js fix/iron-session-fetch | Change |
| - | - | - | - |
| buildDuration | 13.4s | 13.3s | -144ms |
| nodeModulesSize | 66.4 MB | 66.4 MB | ✓ |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | jensmeindertsma/next.js fix/iron-session-fetch | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.392 | 2.31 | -0.08 |
| / avg req/sec | 1044.97 | 1082.33 | +37.36 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.542 | 1.516 | -0.03 |
| /error-in-render avg req/sec | 1621.28 | 1649.54 | +28.26 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | jensmeindertsma/next.js fix/iron-session-fetch | Change |
| - | - | - | - |
| main-HASH.js gzip | 6.63 kB | 6.63 kB | ✓ |
| webpack-HASH.js gzip | 751 B | 751 B | ✓ |
| 19b7e98f51cc..e4a8.js gzip | 10.7 kB | 10.7 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 57.2 kB | 57.2 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | jensmeindertsma/next.js fix/iron-session-fetch | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 5.72 kB | 5.72 kB | ✓ |
| webpack-HASH..dule.js gzip | 751 B | 751 B | ✓ |
| 19b7e98f51cc..dule.js gzip | 7.1 kB | 7.1 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 52.7 kB | 52.7 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | jensmeindertsma/next.js fix/iron-session-fetch | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 26.4 kB | 26.4 kB | ✓ |
| Overall change | 26.4 kB | 26.4 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | jensmeindertsma/next.js fix/iron-session-fetch | Change |
| - | - | - | - |
| _buildManifest.js gzip | 268 B | 268 B | ✓ |
| _buildManife..dule.js gzip | 272 B | 272 B | ✓ |
| Overall change | 540 B | 540 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | jensmeindertsma/next.js fix/iron-session-fetch | Change |
| - | - | - | - |
| index.html gzip | 953 B | 953 B | ✓ |
| link.html gzip | 961 B | 961 B | ✓ |
| withRouter.html gzip | 945 B | 945 B | ✓ |
| Overall change | 2.86 kB | 2.86 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | jensmeindertsma/next.js fix/iron-session-fetch | Change |
| - | - | - | - |
| buildDuration | 14.6s | 14.9s | ⚠️ +343ms |
| nodeModulesSize | 66.4 MB | 66.4 MB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | jensmeindertsma/next.js fix/iron-session-fetch | Change |
| - | - | - | - |
| main-HASH.js gzip | 6.63 kB | 6.63 kB | ✓ |
| webpack-HASH.js gzip | 751 B | 751 B | ✓ |
| 19b7e98f51cc..e4a8.js gzip | 10.7 kB | 10.7 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 57.2 kB | 57.2 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | jensmeindertsma/next.js fix/iron-session-fetch | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 5.72 kB | 5.72 kB | ✓ |
| webpack-HASH..dule.js gzip | 751 B | 751 B | ✓ |
| 19b7e98f51cc..dule.js gzip | 7.1 kB | 7.1 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 52.7 kB | 52.7 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | jensmeindertsma/next.js fix/iron-session-fetch | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 26.4 kB | 26.4 kB | ✓ |
| Overall change | 26.4 kB | 26.4 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | jensmeindertsma/next.js fix/iron-session-fetch | Change |
| - | - | - | - |
| _buildManifest.js gzip | 268 B | 268 B | ✓ |
| _buildManife..dule.js gzip | 272 B | 272 B | ✓ |
| Overall change | 540 B | 540 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | vercel/next.js canary  | jensmeindertsma/next.js fix/iron-session-fetch | Change |
| - | - | - | - |
| _error.js | 876 kB | 876 kB | ✓ |
| 404.html | 4.17 kB | 4.17 kB | ✓ |
| hooks.html | 3.79 kB | 3.79 kB | ✓ |
| index.js | 877 kB | 877 kB | ✓ |
| link.js | 916 kB | 916 kB | ✓ |
| routerDirect.js | 910 kB | 910 kB | ✓ |
| withRouter.js | 910 kB | 910 kB | ✓ |
| Overall change | 4.5 MB | 4.5 MB | ✓ |

</details>

</details>
Commit: b11ec8a9b01eb435a2318ac3fc7e9925a618e11b

**Review by IamLizu — DISMISSED** [review]

File: examples/with-iron-session/package.json [hunk 1]
```diff
@@ -8,11 +8,11 @@
     "start": "next start"
   },
   "dependencies": {
-    "next": "9.3.6",
-    "next-iron-session": "4.0.0",
+    "next": "9.4.4",
```
Comment (Timer) [inline examples/with-iron-session/package.json:11]:
```suggestion
    "next": "^9.4.4",
```

**Review by Timer — APPROVED** [review]

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | jensmeindertsma/next.js fix/iron-session-fetch | Change |
| - | - | - | - |
| buildDuration | 14.2s | 14.1s | -57ms |
| nodeModulesSize | 66.4 MB | 66.4 MB | ✓ |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | jensmeindertsma/next.js fix/iron-session-fetch | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.428 | 2.385 | -0.04 |
| / avg req/sec | 1029.46 | 1048.13 | +18.67 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.623 | 1.568 | -0.05 |
| /error-in-render avg req/sec | 1540.47 | 1594.12 | +53.65 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | jensmeindertsma/next.js fix/iron-session-fetch | Change |
| - | - | - | - |
| main-HASH.js gzip | 6.63 kB | 6.63 kB | ✓ |
| webpack-HASH.js gzip | 751 B | 751 B | ✓ |
| 19b7e98f51cc..e4a8.js gzip | 10.7 kB | 10.7 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 57.2 kB | 57.2 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | jensmeindertsma/next.js fix/iron-session-fetch | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 5.72 kB | 5.72 kB | ✓ |
| webpack-HASH..dule.js gzip | 751 B | 751 B | ✓ |
| 19b7e98f51cc..dule.js gzip | 7.1 kB | 7.1 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 52.7 kB | 52.7 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | jensmeindertsma/next.js fix/iron-session-fetch | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 26.4 kB | 26.4 kB | ✓ |
| Overall change | 26.4 kB | 26.4 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | jensmeindertsma/next.js fix/iron-session-fetch | Change |
| - | - | - | - |
| _buildManifest.js gzip | 268 B | 268 B | ✓ |
| _buildManife..dule.js gzip | 272 B | 272 B | ✓ |
| Overall change | 540 B | 540 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | jensmeindertsma/next.js fix/iron-session-fetch | Change |
| - | - | - | - |
| index.html gzip | 953 B | 953 B | ✓ |
| link.html gzip | 961 B | 961 B | ✓ |
| withRouter.html gzip | 945 B | 945 B | ✓ |
| Overall change | 2.86 kB | 2.86 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | jensmeindertsma/next.js fix/iron-session-fetch | Change |
| - | - | - | - |
| buildDuration | 15.6s | 15.4s | -210ms |
| nodeModulesSize | 66.4 MB | 66.4 MB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | jensmeindertsma/next.js fix/iron-session-fetch | Change |
| - | - | - | - |
| main-HASH.js gzip | 6.63 kB | 6.63 kB | ✓ |
| webpack-HASH.js gzip | 751 B | 751 B | ✓ |
| 19b7e98f51cc..e4a8.js gzip | 10.7 kB | 10.7 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 57.2 kB | 57.2 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | jensmeindertsma/next.js fix/iron-session-fetch | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 5.72 kB | 5.72 kB | ✓ |
| webpack-HASH..dule.js gzip | 751 B | 751 B | ✓ |
| 19b7e98f51cc..dule.js gzip | 7.1 kB | 7.1 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 52.7 kB | 52.7 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | jensmeindertsma/next.js fix/iron-session-fetch | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 26.4 kB | 26.4 kB | ✓ |
| Overall change | 26.4 kB | 26.4 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | jensmeindertsma/next.js fix/iron-session-fetch | Change |
| - | - | - | - |
| _buildManifest.js gzip | 268 B | 268 B | ✓ |
| _buildManife..dule.js gzip | 272 B | 272 B | ✓ |
| Overall change | 540 B | 540 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | vercel/next.js canary  | jensmeindertsma/next.js fix/iron-session-fetch | Change |
| - | - | - | - |
| _error.js | 876 kB | 876 kB | ✓ |
| 404.html | 4.17 kB | 4.17 kB | ✓ |
| hooks.html | 3.79 kB | 3.79 kB | ✓ |
| index.js | 877 kB | 877 kB | ✓ |
| link.js | 916 kB | 916 kB | ✓ |
| routerDirect.js | 910 kB | 910 kB | ✓ |
| withRouter.js | 910 kB | 910 kB | ✓ |
| Overall change | 4.5 MB | 4.5 MB | ✓ |

</details>

</details>
Commit: 43ed86b82c82cdc9c55d5cd9434c06834a6e6156

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | jensmeindertsma/next.js fix/iron-session-fetch | Change |
| - | - | - | - |
| buildDuration | 13.8s | 13.8s | ⚠️ +36ms |
| nodeModulesSize | 66.4 MB | 66.4 MB | ✓ |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | jensmeindertsma/next.js fix/iron-session-fetch | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.429 | 2.403 | -0.03 |
| / avg req/sec | 1029.32 | 1040.55 | +11.23 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.614 | 1.56 | -0.05 |
| /error-in-render avg req/sec | 1548.62 | 1602.17 | +53.55 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | jensmeindertsma/next.js fix/iron-session-fetch | Change |
| - | - | - | - |
| main-HASH.js gzip | 6.63 kB | 6.63 kB | ✓ |
| webpack-HASH.js gzip | 751 B | 751 B | ✓ |
| 19b7e98f51cc..e4a8.js gzip | 10.7 kB | 10.7 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 57.2 kB | 57.2 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | jensmeindertsma/next.js fix/iron-session-fetch | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 5.72 kB | 5.72 kB | ✓ |
| webpack-HASH..dule.js gzip | 751 B | 751 B | ✓ |
| 19b7e98f51cc..dule.js gzip | 7.1 kB | 7.1 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 52.7 kB | 52.7 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | jensmeindertsma/next.js fix/iron-session-fetch | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 26.4 kB | 26.4 kB | ✓ |
| Overall change | 26.4 kB | 26.4 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | jensmeindertsma/next.js fix/iron-session-fetch | Change |
| - | - | - | - |
| _buildManifest.js gzip | 268 B | 268 B | ✓ |
| _buildManife..dule.js gzip | 272 B | 272 B | ✓ |
| Overall change | 540 B | 540 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | jensmeindertsma/next.js fix/iron-session-fetch | Change |
| - | - | - | - |
| index.html gzip | 953 B | 953 B | ✓ |
| link.html gzip | 961 B | 961 B | ✓ |
| withRouter.html gzip | 945 B | 945 B | ✓ |
| Overall change | 2.86 kB | 2.86 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | jensmeindertsma/next.js fix/iron-session-fetch | Change |
| - | - | - | - |
| buildDuration | 15.6s | 15.2s | -364ms |
| nodeModulesSize | 66.4 MB | 66.4 MB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | jensmeindertsma/next.js fix/iron-session-fetch | Change |
| - | - | - | - |
| main-HASH.js gzip | 6.63 kB | 6.63 kB | ✓ |
| webpack-HASH.js gzip | 751 B | 751 B | ✓ |
| 19b7e98f51cc..e4a8.js gzip | 10.7 kB | 10.7 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 57.2 kB | 57.2 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | jensmeindertsma/next.js fix/iron-session-fetch | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 5.72 kB | 5.72 kB | ✓ |
| webpack-HASH..dule.js gzip | 751 B | 751 B | ✓ |
| 19b7e98f51cc..dule.js gzip | 7.1 kB | 7.1 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 52.7 kB | 52.7 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | jensmeindertsma/next.js fix/iron-session-fetch | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 26.4 kB | 26.4 kB | ✓ |
| Overall change | 26.4 kB | 26.4 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | jensmeindertsma/next.js fix/iron-session-fetch | Change |
| - | - | - | - |
| _buildManifest.js gzip | 268 B | 268 B | ✓ |
| _buildManife..dule.js gzip | 272 B | 272 B | ✓ |
| Overall change | 540 B | 540 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | vercel/next.js canary  | jensmeindertsma/next.js fix/iron-session-fetch | Change |
| - | - | - | - |
| _error.js | 876 kB | 876 kB | ✓ |
| 404.html | 4.17 kB | 4.17 kB | ✓ |
| hooks.html | 3.79 kB | 3.79 kB | ✓ |
| index.js | 877 kB | 877 kB | ✓ |
| link.js | 916 kB | 916 kB | ✓ |
| routerDirect.js | 910 kB | 910 kB | ✓ |
| withRouter.js | 910 kB | 910 kB | ✓ |
| Overall change | 4.5 MB | 4.5 MB | ✓ |

</details>

</details>
Commit: b5f147bda80d875e8f14461a967e1f386e710eed

**Review by timneutkens — APPROVED** [review]

**Comment by timneutkens** [comment]
Thanks @jensmeindertsma!

**Pull request merged**